server_structs/xrootd_server.go

```diff
@@ -41,8 +44,49 @@ type (
 	NamespaceHolder struct {
 		namespaceAds []NamespaceAdV2
 	}
+
+	ServerPrefix string // The base namespace prefix for origin/cache server
+)
+
+const (
+	CachePrefix  ServerPrefix = "/caches"
```

Comment:
These should probably contain a trailing `/` to be safe. Otherwise something like
```
prefix := "/cachesfoo"
isCache := strings.HasPrefix(prefix, server_structs.CachePrefix.String())
```
might tell us the prefix indicates a cache. You have functions that do a better job of checking this already, but it's very possible a future developer won't realize they exist and end up checking things like this.